src/main/java/winston/Parser.java

```diff
@@ -0,0 +1,91 @@
+package winston;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.InputMismatchException;
+
+/**
+ * A class that manages user commands.
+ */
+public class Parser {
+    private final TaskList taskList;
+
+    /**
+     * Constructor for class Parser.
+     * 
+     * @param taskList the TaskList that will be modified based on commands.
+     */
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Method that parses the given user input.
+     * 
+     * @param str user input.
+     * @return A command depending on the user input.
+     */
+    public Command parse(String str) {
```

Comment:
I like that all the if and else if blocks are structured nicely, making very easy to trace the code! Good job!! 👍